File: src/main/scala/ru/tinkoff/gatling/amqp/protocol/RabbitMQConnectionFactoryBuilder.scala
package ru.tinkoff.gatling.amqp.protocol

import com.rabbitmq.client.ConnectionFactory

case class RabbitMQConnectionFactoryBuilder(
    host: Option[String] = None,
    port: Option[Int] = None,
    username: Option[String] = None,
    password: Option[String] = None,
    virtualHost: Option[String] = None
) {

  def username(rabbitUsername: String): RabbitMQConnectionFactoryBuilder =
    this.copy(username = Some(rabbitUsername))
  def password(rabbitPassword: String): RabbitMQConnectionFactoryBuilder =
    this.copy(password = Some(rabbitPassword))

  def port(rabbitPort: Int): RabbitMQConnectionFactoryBuilder =
    this.copy(port = Some(rabbitPort))

  def vhost(rabbitVHost: String): RabbitMQConnectionFactoryBuilder =
    this.copy(virtualHost = Some(rabbitVHost))

  def build: ConnectionFactory = {
    val cf = new ConnectionFactory()

    host.foreach(cf.setHost)
    port.foreach(cf.setPort)
    username.foreach(cf.setUsername)
    password.foreach(cf.setPassword)
    virtualHost.foreach(cf.setVirtualHost)

    cf

  }
}

File: src/main/java/ru/tinkoff/gatling/javaapi/protocol/RabbitMQConnectionFactoryBuilder.java
package ru.tinkoff.gatling.javaapi.protocol;

import com.rabbitmq.client.ConnectionFactory;

public class RabbitMQConnectionFactoryBuilder{
    private ru.tinkoff.gatling.amqp.protocol.RabbitMQConnectionFactoryBuilder wrapped;

    public RabbitMQConnectionFactoryBuilder(ru.tinkoff.gatling.amqp.protocol.RabbitMQConnectionFactoryBuilder wrapped){
        this.wrapped = wrapped;
    }

    public RabbitMQConnectionFactoryBuilder username(String rabbitUsername){
        this.wrapped = wrapped.username(rabbitUsername);
        return this;
    }

    public RabbitMQConnectionFactoryBuilder password(String rabbitPassword){
        this.wrapped = wrapped.password(rabbitPassword);
        return this;
    }

    public RabbitMQConnectionFactoryBuilder port(Integer port){
        this.wrapped = wrapped.port(port);
        return this;
    }

    public RabbitMQConnectionFactoryBuilder vhost(String rabbitVHost){
        this.wrapped = wrapped.vhost(rabbitVHost);
        return this;
    }

    public ConnectionFactory build(){
        return wrapped.build();
    }
}
